adapters/repos/db/vector/cache/sharded_lock_cache.go

```diff
@@ -207,6 +211,35 @@ func (s *shardedLockCache[T]) MultiGet(ctx context.Context, ids []uint64) ([][]T
 	return out, errs
 }
 
+func (s *shardedLockCache[T]) GetAllInCurrentLock(ctx context.Context, id uint64, out [][]T, errs []error) ([][]T, []error, uint64, uint64) {
+	start := (id / s.shardedLocks.PageSize) * s.shardedLocks.PageSize
+	end := start + s.shardedLocks.PageSize
+
+	if end > uint64(len(s.cache)) {
+		end = uint64(len(s.cache))
+	}
+
+	s.shardedLocks.RLock(start)
+	defer s.shardedLocks.RUnlock(start)
+
+	for i := start; i < end; i++ {
+		vec := s.cache[i]
+
+		if vec == nil {
+			vecFromDisk, err := s.handleCacheMiss(ctx, i, false)
```

Comment:
It isn't valid that we skip the locking here because the cache is only protected by an `RLock` while if we handle the cache miss we need to upgrade to `Lock`.